### ISSUE TITLE
Dependency problem or ??

### ISSUE THREAD

**Issue by jnmichaud** [andris9/Nodemailer#402]
I'm fairly clueless as to what I actually found here. This could be my own fault, but I'm really not sure where else to ask, so I apologize in advance if so.

My app is throwing this error when I try to require nodemailer:

util.js:538
  ctor.prototype = Object.create(superCtor.prototype, {
                                          ^
TypeError: Cannot read property 'prototype' of undefined
    at Object.exports.inherits (util.js:538:43)
    at Object.<anonymous> (/home/e6serv/e6Code/e6Inner/node_modules/nodemailer/node_modules/libmime/node_modules/libbase64/lib/libbase64.js:82:6)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:362:17)
    at require (module.js:378:17)
    at Object.<anonymous> (/home/e6serv/e6Code/e6Inner/node_modules/nodemailer/node_modules/libmime/src/libmime.js:4:17)
    at Module._compile (module.js:449:26)

Any suggestions?

**Comment by andris9** [comment]
You are probably using Node v0.8 while latest Nodemailer requires Node v0.10+. For Node 0.8 use older Nodemailer 0.x branch in your dependencies like this:

    "dependencies": {
        "nodemailer": "~0.7.0"
    }

And here are the [docs for the 0.x branch](https://github.com/andris9/Nodemailer/blob/0.7/README.md)

**Comment by jnmichaud** [comment]
I'm using Node 0.12.0

**Comment by andris9** [comment]
Are you sure? This error line:

    TypeError: Cannot read property 'prototype' of undefined
        at Object.exports.inherits (util.js:538:43)

Only happens in Node v0.8 while running `util.inherit(func, undefined_value)`. If you do the same in Node v0.12, then the line number should be *util.js:634:43* not *util.js:538:43*

**Comment by jnmichaud** [comment]
Yeah I'm positive. I just upgraded it today.

node -v returns
v0.12.0

Now I'm really confused.

**Comment by andris9** [comment]
What exactly causes the error with Nodemailer can be reproduced with this one-liner:

    require('util').inherits(function(){}, require('stream').Transform);

Nodemailer tries to inherit from the stream.Transform constructor but stream.Transform is not available in Node 0.8.

```
~ $ node -v
v0.8.28
~ $ node
> require('util').inherits(function(){}, require('stream').Transform);
TypeError: Cannot read property 'prototype' of undefined
    at Object.exports.inherits (util.js:538:43)
    at repl:1:17
    at REPLServer.self.eval (repl.js:109:21)
    at rli.on.self.bufferedCmd (repl.js:258:20)
    at REPLServer.self.eval (repl.js:116:5)
    at Interface.<anonymous> (repl.js:248:12)
    at Interface.EventEmitter.emit (events.js:96:17)
    at Interface._onLine (readline.js:200:10)
    at Interface._line (readline.js:518:8)
    at Interface._ttyWrite (readline.js:736:14)
> 
```

If you do the same in Node v0.10 or v0.12 you get no error:

```
~ $ node -v
v0.12.0
~ $ node
> require('util').inherits(function(){}, require('stream').Transform);
undefined
>
```

**Comment by jnmichaud** [comment]
I appreciate the help, andris.

Apparently the file at /usr/bin/node (with which I am executing my node.js apps) is not node v0.12.0 even though the version the machine is running currently IS 0.12.0

I shall attempt to untangle my mess. Sorry for the trouble, but you WERE incredibly helpful. I had no idea what was going on.
